python/semantic_kernel/connectors/ai/anthropic/services/anthropic_chat_completion.py

```diff
@@ -164,20 +169,59 @@ async def _inner_get_streaming_chat_message_contents(
                         author_role = stream_event.message.role
                         metadata["usage"]["input_tokens"] = stream_event.message.usage.input_tokens
                         metadata["id"] = stream_event.message.id
+
                     elif isinstance(stream_event, (RawContentBlockDeltaEvent, RawMessageDeltaEvent)):
+                        if hasattr(stream_event, "index") and stream_event.index is not None:
+                            content_block_idx = stream_event.index
                         yield [
                             self._create_streaming_chat_message_content(
                                 stream_event, content_block_idx, author_role, metadata
                             )
                         ]
-                    elif isinstance(stream_event, ContentBlockStopEvent):
-                        content_block_idx += 1
         except Exception as ex:
             raise ServiceResponseException(
                 f"{type(self)} service failed to complete the request",
                 ex,
             ) from ex
 
+    def _prepare_chat_history_for_request(
+        self,
+        chat_history: "ChatHistory",
+        role_key: str = "role",
+        content_key: str = "content",
+    ) -> tuple[list[dict[str, Any]], str | None]:
+        """Prepare the chat history for an Anthropic request.
+
+        Allowing customization of the key names for role/author, and optionally overriding the role.
+
+        Args:
+            chat_history: The chat history to prepare.
+            role_key: The key name for the role/author.
+            content_key: The key name for the content/message.
+
+        Returns:
+            prepared_chat_history:
+            A tuple containing the prepared chat history and the first SYSTEM message content.
+        """
+        system_message_content = None
+        remaining_messages = []
+        system_message_found = False
+
+        for message in chat_history.messages:
+            # Skip system messages after the first one is found
+            if message.role == AuthorRole.SYSTEM and not system_message_found:
+                system_message_content = message.content
+                system_message_found = True
+                continue
+
+            # Include only role and content keys for the remaining messages
+            remaining_messages.append({
+                role_key: getattr(message, role_key),
+                content_key: getattr(message, content_key),
+            })
```

Comment:
Do we need to filter any additional system messages out of the remaining messages?

Reply:
Yeah good catch, we had just chatted about it, and I didn't include that.

Reply:
Can we have a test that starts with a chat history with multiple system messages?